### ISSUE TITLE
Data augmentation code does not include identity scale factor

### ISSUE THREAD

**Issue by ishmael1985** [twtygqyy/pytorch-vdsr#17]
I examined the Matlab code in data/generate_train.m and it seems that the augmentation does not include the combination of transforms (flip, rotate, downsample) with the original image size (scale=1). While your results are already good enough, would missing out scale=1 impact the result considerably?

**Comment by twtygqyy** [comment]
Hi @ishmael1985 adding scale=1 in the data generation part will not affect the PSNR result, since the networks will not enhance the images with good quality.

**Comment by ishmael1985** [comment]
@twtygqyy I do not get it. Why can't the original unscaled images and their augmented transformed versions (flip, rotate) be part of the overall training dataset? When downscaling, we are losing important detail of the original images which should be learned by the network. The Matlab code in generate_train.m indicates that the original images are not being used as part of the training dataset. Wouldn't including those unscaled images help improve the overall PSNR result?
Unless I am overlooking something in the Matlab code, the original unscaled images (scale=1) and their flipped, rotated transformed versions are not included.

**Comment by twtygqyy** [comment]
@ishmael1985 Sorry, did you mean using unscaled images as both inputs and outputs, and each input-output pair is exactly the same?

**Comment by ishmael1985** [comment]
Yep, somehow your code doesn't include them as part of hdf5 packing for the preprocessed training dataset. I may have overlooked something going through all those nested for loops, but it seems you are missing out the unscaled image pairs.

**Comment by twtygqyy** [comment]
@ishmael1985 According to my experiments, putting unscaled images cannot help to improve the performance, and sometime will even leads to vanishing gradient due to the residual learning. (giving all zero to the weights will output the best result)

**Comment by ishmael1985** [comment]
I see. So basically the original unscaled image (downsize=1) and downsized versions (0.7, 0.5) are excluded from the training dataset, is that correct? Only the scaled versions (1/2, 1/3, 1/4) of the downsized images (1, 0.7, 0.5) will be used for the training dataset then?

**Comment by twtygqyy** [comment]
For the input samples, yes. But the size of the samples are always the same as the labels, see `im_input = imresize(imresize(im_label,1/scale(s),'bicubic'),[hei,wid],'bicubic');`

**Comment by ishmael1985** [comment]
Thanks. This is quite important for me to know since you've managed to reproduce the results and I'd like to take your word on it, as I'm working on a generic data augmentation script which I'd like to verify with your training script. For me to replicate your results, I'll have to understand your Matlab script so that I can follow its behaviour in the data augmentation script I'm currently developing.
